samples/backtest_grid.py

```diff
@@ -0,0 +1,329 @@
+        self._buy_orders_ids: List[str] = []
+        self._sell_orders_ids: List[str] = []
+
+    async def on_trading_signal(self, trading_signal: BaseTradingSignal):
+        """
+        Handle trading signals
+
+        @param trading_signal: TradingSignal instance
+        """
+        pairs = list(trading_signal.get_pairs())
+        logging.info(StructuredMessage("Trading signal", pairs=pairs))
+
+        try:
+            coros = []
+            for pair, target_position in pairs:
+                if self._borrowing_disabled and target_position == bs.Position.SHORT:
+                    target_position = bs.Position.NEUTRAL
+                coros.append(self.switch_position(pair, target_position))
+            await asyncio.gather(*coros)
+        except Exception as e:
+            logging.exception(e)
+
+    async def switch_position(self, pair: bs.Pair, target_position: bs.Position, force: bool = False):
+        """
+        Handle change in trading position (SHORT, LONG, NEUTRAL)
+
+        @param pair: Pair instance
+        @param target_position: New position
+        @param force: True if user want to execute even if old position == new position
+        @return:
+        """
+        current_pos_info = await self.get_position_info(pair)
+
+        # Unless force is set, we can ignore the request if we're already there.
+        if not force and any([
+            # current_pos_info is None and target_position == bs.Position.NEUTRAL,
+            (
+                    current_pos_info is not None
+                    and current_pos_info == target_position
+            )
+        ]):
+            return
+        self._positions[pair] = PositionInfo(pair, target_position)
+
+        if target_position != bs.Position.NEUTRAL:
+            return
+
+        # SIGNAL NEUTRAL (BBands cross the middle)
+        buy_order_ids = [order.id for order in await self._exchange.get_open_orders(pair) if
+                         order.operation == OrderOperation.BUY]
+
+        if len(buy_order_ids) == 0:
+
+            (bid, ask), pair_info = await asyncio.gather(
+                self._exchange.get_bid_ask(pair),
+                self._exchange.get_pair_info(pair),
+            )
+            order_size = bs.truncate_decimal(Decimal(self._position_amount), pair_info.base_precision)
+            grid_step_from_pips = Decimal(self._grid_step * Point(pair).value)
+
+            # place initial buy limit orders
+            for i in range(self._grid_size):
+                price = bs.truncate_decimal(Decimal(bid - (grid_step_from_pips * (i + 1))), pair_info.quote_precision)
+                if self._martingale_sizes is True:
+                    order_size = bs.truncate_decimal(Decimal(self._position_amount * (i + 1)), pair_info.base_precision)
+                operation = bs.OrderOperation.BUY
+                logging.info(
+                    StructuredMessage("Creating market order", operation=operation, pair=pair, order_size=order_size,
```

Comment:
Should be *Creating limit order* instead. In appears in multiple places within this file.

Reply:
Ok, will fix it.